### DISCUSSION TITLE
Nested DTO not deserialized

### DISCUSSION THREAD

**Discussion by MichaelKubovic** [api-platform/core#5160] [Q&A]
**API Platform version(s) affected**: 2.5.3

**Description**  

Having a resource that uses DTO for input and that class contains property that contains array of DTOs of other class.

Generated documentation says that the property has type of array of strings.

Sending a request actually makes the data of the nested dto available as an associative array, although I'd expect it to be deserialized into an object.

**How to reproduce**  

```php
/**
 * @ApiResource(
 *     collectionOperations={
 *          "post"={
 *              "status"=202,
 *              "input"=App\Command\CreateFoo::class,
 *              "output"=false
 *          },
 *     },
 *     itemOperations={},
 *     messenger="input",
 *     output=false,
 * )
 */
class TestResource
{
    // not important
    public $foo;
}
```

```php
final class CreateFoo
{
   // not important
    public $name;

    /** @var CreateBar[] */
    public $bars = [];
}
```

```php
final class CreateBar
{
    public $baz;
}
```

**Possible Solution**  

**Additional Context**

**Comment by soyuka** [comment]
Would you be able to debug this a bit more? The denormalization of this is done here:

https://github.com/api-platform/core/blob/5fa9b567dd2b86cf40ff6f8784776535409bb321/src/Serializer/AbstractItemNormalizer.php#L189 

It'd be interesting to know what it does when it encounters `/** @var CreateBar[] */` I'm not sure that the denormalizer is aware that it is a collection of objects (doesn't read php docs).

**Comment by teohhanhui** [comment]
We have no way to know how to denormalize those objects. You'd have to do this in your `DataTransformer`.

**Comment by Kaspiman** [comment]
@teohhanhui I have the same issue (array of strings instead of array of DTO's) and wrote comment in this issue https://github.com/api-platform/api-platform/issues/1202#issuecomment-573676290 with some debug experience.

Offtop: can you explain: all issues from `template` are not relevant now?

**Comment by Kaspiman** [comment]
@MichaelKubovic @teohhanhui I found the solution, you have to configure DocumentationNormalizer like this:

```
    api_platform.swagger.normalizer.documentation:
        arguments:
            $jsonSchemaTypeFactory: '@api_platform.json_schema.type_factory' # here is missing service
```
